File: backend-springboot/src/main/java/com/cosmetics/model/Banner.java
package com.cosmetics.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "banners")
public class Banner {

    @Id
    private String id;

    private String title;

    @Field("image_url")
    private String imageUrl;

    @Field("link_url")
    private String linkUrl;

    @Builder.Default
    private String position = "hero"; // hero, popup, sidebar, category_top

    @Field("sort_order")
    @Builder.Default
    private Integer sortOrder = 0;

    @Field("is_active")
    @Builder.Default
    private Boolean isActive = true;

    @Field("start_date")
    private LocalDateTime startDate;

    @Field("end_date")
    private LocalDateTime endDate;
}

File: backend-nodejs/src/models/Banner.js
import mongoose from "mongoose";

const bannerSchema = new mongoose.Schema(
  {
    title: { type: String, required: true },
    image_url: { type: String, required: true },
    link_url: { type: String, default: "" },
    position: { type: String, default: "hero" },
    sort_order: { type: Number, default: 0 },
    is_active: { type: Boolean, default: true },
    start_date: { type: Date, default: null },
    end_date: { type: Date, default: null },
  },
  { timestamps: { createdAt: "created_at", updatedAt: "updated_at" } },
);

bannerSchema.set("toJSON", {
  transform: (doc, ret) => {
    ret.id = ret._id;
    ret.imageUrl = ret.image_url;
    ret.linkUrl = ret.link_url;
    ret.sortOrder = ret.sort_order;
    ret.isActive = ret.is_active;
    ret.startDate = ret.start_date;
    ret.endDate = ret.end_date;
    delete ret._id;
    delete ret.__v;
    // Xóa các trường snake_case cũ
    delete ret.image_url;
    delete ret.link_url;
    delete ret.sort_order;
    delete ret.is_active;
    delete ret.start_date;
    delete ret.end_date;
    return ret;
  },
});

export default mongoose.model("Banner", bannerSchema);
